plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -169,19 +173,50 @@ public static void installAppEngineFacet(IFacetedProject facetedProject,
     if (facetedProject.hasProjectFacet(FACET)) {
       return;
     }
+
+    final IFacetedProjectWorkingCopy fpjwc = facetedProject.createWorkingCopy();
+    fpjwc.detect(subMonitor.newChild(20));
+    fpjwc.commitChanges(subMonitor.newChild(20));
+
+    if (facetedProject.hasProjectFacet(FACET)) {
+      // success!
+      return;
+    }
+
+    Set<IProjectFacet> previousFixedFacets = fpjwc.getFixedProjectFacets();
     FacetUtil facetUtil = new FacetUtil(facetedProject);
-    facetUtil.addFacetToBatch(FACET_VERSION, null /* config */);
+    // See if the default AppEngine Standard facet is ok
+    if (!FacetUtil.conflictsWith(fpjwc, FACET.getDefaultVersion())) {
+      facetUtil.addFacetToBatch(FACET.getDefaultVersion(), null);
+      fpjwc.addProjectFacet(FACET.getDefaultVersion());
```

Comment:
We're using `FacetUtil`, so I think there's no point of using `fpjwc`? Moreover, there is no `commitChanges()` onward, so  I think `fpjwc` will have no effect at all.